### DISCUSSION TITLE
Gamepad4 and multi-report Gamepad use

### DISCUSSION THREAD

**Discussion by alijani1** [NicoHood/HID#254] [Q&A]
@NicoHood, thanks for the great library.   I am using a pro micro (32u4) and as I understand, it is limited to 6 endpoints and since CDC Serial uses 3, we will only be able to use 3 remaining for single report devices.  From what I see in your classes and examples, there is a Gamepad class and Gamepad1,2,3,4 single report classes.

I have several questions and I would be grateful if you can share your thoughts on them:

1. First question is why is there a Gamepad4 if the maximum of 3? I tested it and gamepad4 class done not function.  is there a way to use gamepad4 some other way?  I am trying to have 4 game pads on single usb, ideally each on its own end point.
2. I have some situations where I only want to enable Gamepad1 and Gamepad2 but somehow the OS is always seeing 3 devices.  I only do Gamepad1.begin() and Gamepad2.begin().  I even tried Gamepad3.end() but I am unable to hide it.
3. If I wanted to use a single end point with Multi-report, to support 4 gamepads, how is it done?  I don't see any methods in the Gamepad class to control report id or multiple hids.   If there any example I can find on this?

Thanks in advance and again for a super library. Its the only one I have found that works well even with MacOs!

**Comment by NicoHood** [comment]
1. Good Question. I guess you are right... why did I implement that? I guess you'd need to disable the CDC serial, but that is not that trivial. Otherwise you'd need something like Lufa and code that low level
2. Try to remove all references for Gamepad3. If it is references, it will be compiled. Otherweise I'd recommend [this](https://github.com/NicoHood/HID/wiki/Troubleshoot-FAQ#switching-the-hid-core), but since you are on MAC that might not be helpful.
3. Another good question. As far as I remember, the single report did not work with windows and more than one Gamepad. I guess that was the reason.

**Reply by alijani1** [reply]
Thanks. 

On #1/#2, I will just comment out endpoints not needed.
On #3, Windows 10 does support multi-report endpoints and so does linux,  I was using MHeironimus/ArduinoJoystickLibrary and its working fine with 4 gamepads on single endpoint.  Main reason I started to look at yours instead was because I was having some trouble with Mac OS and your library actually worked well on Mac but I am stuck because I cannot get to 4 gamepads I need. I like your efficient code much better. Anything you can suggest on how to use your library for multiple gamepads on a single endpoint?

**Reply by NicoHood** [reply]
Checkout https://github.com/arduino/Arduino/issues/6387#issuecomment-309363314 on how to disable cdc serial. Also interesting: https://github.com/NicoHood/HID/issues/225 If that works, please let me know.

Using Gamepads with multireport? I guess you'd need to implement that. Feel free to do that.
